plugins/org.eclipse.fordiac.ide.ui/src/org/eclipse/fordiac/ide/ui/contentoutline/MultiPageSite.java

```diff
@@ -0,0 +1,252 @@
+		serviceLocator.registerService(IPageSiteHolder.class, (IPageSiteHolder) () -> MultiPageSite.this);
+
+		final IHandlerService handlerService = new LegacyHandlerService(e4Context);
+		e4Context.set(IHandlerService.class, handlerService);
+
+		e4Context.set(IContextService.class.getName(), new ContextFunction() {
+			@Override
+			public Object compute(final IEclipseContext context, final String contextKey) {
+				if (contextService == null) {
+					contextService = new NestableContextService(context.getParent().get(IContextService.class),
+							new ActivePartExpression(partSite.getPart()));
+				}
+				return contextService;
+			}
+		});
+	}
+
+	/**
+	 * Disposes of the menu extender contributions.
+	 */
+	protected void dispose() {
+		if (menuExtenders != null) {
+			final HashSet<MenuManager> managers = HashSet.newHashSet(menuExtenders.size());
+			for (final PopupMenuExtender menuExtender : menuExtenders) {
+				final PopupMenuExtender ext = menuExtender;
+				managers.add(ext.getManager());
+				ext.dispose();
+			}
+			if (!managers.isEmpty()) {
+				for (final MenuManager mgr : managers) {
+					mgr.dispose();
+				}
+			}
+			menuExtenders = null;
+		}
+		subActionBars.dispose();
+
+		if (contextService != null) {
+			contextService.dispose();
+		}
+
+		serviceLocator.dispose();
+		e4Context.dispose();
+	}
+
+	/**
+	 * The PageSite implementation of this <code>IPageSite</code> method returns the
+	 * <code>SubActionBars</code> for this site.
+	 *
+	 * @return the subactionbars for this site
+	 */
+	@Override
+	public SubActionBars getActionBars() {
+		return subActionBars;
+	}
+
+	@Override
+	public <T> T getAdapter(final Class<T> adapter) {
+		return Platform.getAdapterManager().getAdapter(this, adapter);
+	}
+
+	@Override
+	public IWorkbenchPage getPage() {
+		return parentSite.getPage();
+	}
+
+	@Override
+	public ISelectionProvider getSelectionProvider() {
+		return selectionProvider;
+	}
+
+	@Override
+	public final <T> T getService(final Class<T> key) {
+		final T service = serviceLocator.getService(key);
+		if (active && service instanceof final INestable nestable) {
+			nestable.activate();
+		}
+		return service;
+	}
+
+	@Override
+	public Shell getShell() {
+		return parentSite.getShell();
+	}
+
+	@Override
+	public IWorkbenchWindow getWorkbenchWindow() {
+		return parentSite.getWorkbenchWindow();
+	}
+
+	@Override
+	public final boolean hasService(final Class<?> key) {
+		return serviceLocator.hasService(key);
+	}
+
+	@Override
+	public void registerContextMenu(final String menuID, final MenuManager menuMgr,
+			final ISelectionProvider selProvider) {
+		if (menuExtenders == null) {
+			menuExtenders = new ArrayList<>(1);
+		}
+		PartSite.registerContextMenu(menuID, menuMgr, selProvider, false, partSite.getPart(), e4Context, menuExtenders);
+	}
+
+	@Override
+	public void setSelectionProvider(final ISelectionProvider provider) {
+		selectionProvider = provider;
+	}
+
+	/* Package */IEclipseContext getSiteContext() {
```

Comment:
what is the reason for this comment?

Reply:
That was a leftover from the original `PageSite`, removed now.